godot-core/src/builtin/packed_array.rs

```diff
@@ -0,0 +1,544 @@
+                assert!(
+                    index <= len,
+                    "Array insertion index {} is out of bounds: length is {}",
+                    index,
+                    len
+                );
+                self.as_inner().insert(to_i64(index), Self::into_arg(value));
+            }
+
+            /// Removes and returns the element at the specified index. Similar to `remove_at` in
+            /// GDScript, but also returns the removed value.
+            ///
+            /// On large arrays, this method is much slower than `pop_back` as it will move all the array's
+            /// elements after the removed element. The larger the array, the slower `remove` will be.
+            ///
+            /// # Panics
+            ///
+            /// If `index` is out of bounds.
+            // Design note: This returns the removed value instead of `()` for consistency with
+            // `Array` and with `Vec::remove`. Compared to shifting all the subsequent array
+            // elements to their new position, the overhead of retrieving this element is trivial.
+            pub fn remove(&mut self, index: usize) -> $Element {
+                self.check_bounds(index);
+                let element = self.get(index);
+                self.as_inner().remove_at(to_i64(index));
+                element
+            }
+
+            /// Assigns the given value to all elements in the array. This can be used together
+            /// with `resize` to create an array with a given size and initialized elements.
+            pub fn fill(&mut self, value: $Element) {
+                self.as_inner().fill(Self::into_arg(value));
+            }
+
+            /// Appends another array at the end of this array. Equivalent of `append_array` in
+            /// GDScript.
+            pub fn extend_array(&mut self, other: &$PackedArray) {
+                self.as_inner().append_array(other.clone());
+            }
+
+            /// Reverses the order of the elements in the array.
+            pub fn reverse(&mut self) {
+                self.as_inner().reverse();
+            }
+
+            /// Sorts the elements of the array in ascending order.
+            // Presumably, just like `Array`, this is not a stable sort so we might call it
+            // `sort_unstable`. But Packed*Array elements that compare equal are always identical,
+            // so it doesn't matter.
+            pub fn sort(&mut self) {
+                self.as_inner().sort();
+            }
+
+            /// Asserts that the given index refers to an existing element.
+            ///
+            /// # Panics
+            ///
+            /// If `index` is out of bounds.
+            fn check_bounds(&self, index: usize) {
+                let len = self.len();
+                assert!(
+                    index < len,
+                    "Array index {} is out of bounds: length is {}",
+                    index,
+                    len
+                );
```

Comment:
Suggestion:
```rs
                assert!(
                    index < len,
                    "Array index {index} is out of bounds: length is {len}"
                );
```

Does clippy not complain here? I [just fixed](https://github.com/godot-rust/gdextension/pull/86/commits/62cd271c28bfa090c5acaa5589fc5d13c0c7a640) a ton of similar issues 😀

A few other places like this...

Reply:
Not in CI at least... maybe it needs a Rust version upgrade?

Reply:
I'm on 1.66.0 stable and not getting complaints when I run `cargo clippy` in the project root :shrug: I'll fix these ones but we might need a more structural solution here.

Reply:
I encountered them e.g. [in this CI run](https://github.com/godot-rust/gdextension/actions/runs/4018469919/jobs/6904128557), but it was `format!` in that case.

The [Clippy lint doc](https://rust-lang.github.io/rust-clippy/master/index.html#uninlined_format_args) does not explicitly mention `assert!`, but clicking on _Related issues_ shows some issues mentioning `assert!`. Hmm... 🤔